### DISCUSSION TITLE
How do I append row to database?

### DISCUSSION THREAD

**Discussion by egeres** [lastorel/pytion#64] [Announcements]
Accessing databases is pretty straightforward:

```python
...
no   = Notion(token)
db   = no.databases.get(id)
```
And if we run `db.db_query().obj.array`, we can iterate over the entries in a notion database as well as see the properties of each object accessing `.properties` retrieves a dictionary with all the info... but I can't find a way to insert a new page in the database? I would have expected the following code to work, but it doesn't:

```python
new_page = Page.create(
    parent = LinkTo(from_object=db.obj),
    properties = {
        "Name": PropertyValue.create("title", "My title")
    }
)
created_page = db.page_create(new_page)
```

**Comment by lastorel** [comment]
Hello, yes I can see that you execute `page_create` on `db` object.
It works only with `no.pages.` at the moment.
In your case:
```python
new_page = Page.create(
    parent = LinkTo(from_object=db.obj),
    properties = {
        "Name": PropertyValue.create("title", "My title")
    }
)
created_page = no.pages.page_create(new_page)
```
The Page object already has parent attr, because it's mandatory. And there is a constrain to use page_create on specific Element objects. Otherwise it would be possible to execute db.page_create(new_page) and db2.page_create(new_page) that creates two identical pages in the same database
Another way to append to database described here: https://github.com/lastorel/pytion/discussions/62

**Reply by egeres** [reply]
Works!! Thanks!!

**Reply by egeres** [reply]
And um, as a minor comment, I find it a bit counterintuitive that the way to delete a page is to call `.page_update` with `archived=True`, I guess a `.page_delete` method would make more sense? I'm not deep into the logic of notion's API, so maybe the current design follows their protocol more closely

**Comment by lastorel** [comment]
Converting to discussion
